aopy/analysis.py

```diff
@@ -856,3 +854,179 @@ def linear_fit_analysis2D(xdata, ydata, weights=None, fit_intercept=True):
+        false_alarm_prob (float, optional): Desired false alarm probability. Defaults to 0.05.
+
+    Returns:
+        (float): threshold proportion that yields the desired false alarm probability
+    '''
+
+    thresh_props = np.arange(0, 1, step_size)
+    fa_rates = []    
+    for tp in thresh_props:
+        accLLR, selection_time_idx = calc_activity_onset_accLLR(cond2_test, cond1_train, cond2_train, modality, bin_width, thresh_proportion=tp, trial_average=False)
+        accLLR_cond1_within_time = np.logical_and(accLLR > 0, selection_time_idx < cond1_train.shape[0])
+        n_accllr_cond1_within_time = np.count_nonzero(accLLR_cond1_within_time)
+        false_alarms = n_accllr_cond1_within_time / len(cond2_test)
+        fa_rates.append(false_alarms)
+
+    fa_rates_above_desired = thresh_props[np.array(fa_rates) > false_alarm_prob]
+    if not any(fa_rates_above_desired):
+        return fa_rates
+
+    return fa_rates_above_desired[0]
+
+
+def accLLR_wrapper(data_cond1, data_cond2, modality, bin_width, train_prop_input=0.7, step_size=0.01, false_alarm_prob=0.05):
+    '''
+    1. Separates data into 'model building' (training + test) and 'implementing' groups by trials 
+        Assumes model building dataset is split into a 60/40 split
+    2. Calculates accLLR threshold on training and test data
+    3. Implements accLLR on 'implementing' data
+    
+    Args:
+        data_cond1 (npts, ntrials):
+        data_cond2 (npts, ntrials):
+        modality (str): either 'spikes' or 'lfp'
+        train_prop (float): proportion of trials to build the model with
+        
+    '''
+    # Split into model building and testing data sets (outputs (ntrial, npt) datasets)
+    test_data_cond1, build_data_cond1, test_data_cond2, build_data_cond2 = model_selection.train_test_split(data_cond1.T, data_cond2.T, test_size=train_prop_input)
+
+    # Separate model building data into valid/test sets
+    nbuild_trials = build_data_cond1.shape[0]
+    ntrain_trials = np.floor(nbuild_trials*0.6) # not sure if we should have the user input this
+    nvalid_trials = nbuild_trials-ntrain_trials
+
+    train_trialidxs = np.random.choice(np.arange(nbuild_trials), size=ntrain_trials)
+    train_trial_mask = np.zeros(nbuild_trials, dtype=bool)
+    train_trial_mask[train_trialidxs] = True
+
+    train_data_cond1 = build_data_cond1[train_trial_mask,:]
+    train_data_cond2 = build_data_cond2[train_trial_mask,:]
+    valid_data_cond1 = build_data_cond1[np.logical_not(train_trial_mask),:]
+    valid_data_cond2 = build_data_cond2[np.logical_not(train_trial_mask),:]
+
+    # Input train and validate data to calculate acclllr threshold (transpose data back to input into accllr functions)
+    accllr_thresh = calc_accLLR_threshold(train_data_cond1.T, train_data_cond2.T, valid_data_cond2.T, modality, bin_width, step_size=step_size, false_alarm_prob=false_alarm_prob)
```

Comment:
Should I be inputting (npt, ntrial) training data or trial average here to determine the training data to input

Reply:
should just be a single trial. i hadn't considered trial averaging but that would make sense, is that what they do in the paper?